src/Instrumentation/Psr15/src/Psr15Instrumentation.php

```diff
@@ -85,14 +85,14 @@ public static function register(): void
                 if (!$root && $request) {
                     //create http root span
                     $parent = Globals::propagator()->extract($request->getHeaders());
+                    $parent = $parent->with(Psr15Instrumentation::$rootSpanExists, true);
```

Comment:
this looks like it should be storing the root span itself, and doing it after the span is created?

Reply:
ok